src/proxy/http2/Http2ClientSession.cc

```diff
@@ -191,16 +191,22 @@ Http2ClientSession::main_event_handler(int event, void *edata)
     break;
 
   case VC_EVENT_ACTIVE_TIMEOUT:
-  case VC_EVENT_INACTIVITY_TIMEOUT:
+  case VC_EVENT_INACTIVITY_TIMEOUT: {
+    if (this->connection_state.get_shutdown_state() == HTTP2_SHUTDOWN_NONE) {
+      this->connection_state.set_shutdown_state(HTTP2_SHUTDOWN_INITIATED, Http2ErrorCode::HTTP2_ERROR_NO_ERROR);
```

Comment:
I may not understand the issue, but the state should not be changed directly here. It should be set by HTTP2_SESSION_EVENT_SHUTDOWN_INIT event handler. Changing the state at multiple places is going to be super confusing.

Reply:
Will sending this with the SHUTDOWN_CONT event below trigger SHUTDOWN_INIT?  Or are you suggesting that this case should send the SHUTDOWN_INIT event instead and rely on that flow to set the appropriate state?

Reply:
I was suggesting scheduling SHUTDOWN_INIT event instead. But if canceling grace shutdown works we don't need it.